src/pond/lib/timeseries.js

```diff
@@ -344,12 +344,15 @@ class TimeSeries {
     /**
      * Sets a new underlying collection for this TimeSeries.
      *
-     * @param {Collection}  collection The new collection
+     * @param {Collection}  collection       The new collection
+     * @param {boolean}     isChronological  Causes the chronological
+     *                                       order of the events to
+     *                                       not be checked
      *
-     * @return {TimeSeries}            A new TimeSeries
+     * @return {TimeSeries}                  A new TimeSeries
      */
-    setCollection(collection) {
-        if (!collection.isChronological()) {
+    setCollection(collection, isChronological = false) {
+        if (!isChronological && !collection.isChronological()) {
```

Comment:
No, if `isChronological` is false then we do the check. To me we should do the check if `isChronological` is true -- meaning we expect it to be in order already.